specification/attestation.adoc

```diff
@@ -1,309 +1,874 @@
+specification.
+* *Component signer*: This is the hash value of a signing authority for the
+component.
+* *Hash algorithm identifier*: This describes which algorithm was used to generate the
+component measurement. It is recommended to use one of the IANA defined
+<<HashAlgorithmNames>>.
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "Platform Software Components Claim"]
+[source,cddl]
+----
+platform-sw-components-label = TBD
+riscv-ap-tee-sw-component = {
+   1 => text                   ; Component type
+   2 => riscv-ap-tee-hash-type ; Component measurement value
+   3 => text                   ; Component Secure Version Number (SVN)
+ ? 4 => riscv-ap-tee-hash-type ; Component manifest hash
+   5 => riscv-ap-tee-hash-type ; Component signer public key hash value
+   6 => text                   ; Hash algorithm identifier
+}
+
+platform-sw-components = (
+   platform-sw-components-label => [ + riscv-ap-tee-sw-component ]
+)
+----
+
+===== TSM Token
+
+The TSM Token is a nested EAT token in the Evidence and contains a map of
+Claims. The map is wrapped in a `COSE_Sign1` envelope and composed of the
+following claims:
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "TSM Claims Map"]
+[source,cddl]
+----
+tsm-token = #6.61(tsm-token-map)
+
+tsm-token-map = {
+   tsm-challenge     ; Hash of the TVM public key
+   tsm-public-key    ; TSM public key
+   tsm-sw-components ; TSM SW components
+}
+----
+
+All above described Claims must be present in the TSM Token.
+
+====== TSM Challenge Claim
+
+The TSM challenge claim is the hash value of the TVM public key.
+The TVM public key is a TVM guest generated public key, and passed to the TSM as
+an argument to the `sbi_tee_guest_get_evidence()` SBI call.
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "TSM Challenge Claim"]
+[source,cddl]
+----
+tsm-challenge-label = 10 ; EAT nonce
+
+tsm-challenge = (
+   tsm-challenge-label => riscv-ap-tee-hash-type
+)
+----
+
+====== TSM Public Key Claim
+
+The TSM public key claim identifies the platform public key that signs the
+TVM token. The value of the claim is a COSE_Key <<COSE>> structure encoded as a
+CBOR bytes string.
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "TSM Public Key Claim"]
+[source,cddl]
+----
+tsm-public-key-label = TBD
+
+tsm-public-key = (
+   tsm-public-key-label => riscv-ap-tee-public-key-type
+)
+----
+
+====== TSM Software Components Claim
+
+The TSM software components claim lists all software components that compose the
+AP-TEE TSM TCB.
+
+The TSM software components that influence the TSM TCB are the TSM-Driver and the
+TSM.
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "TSM Software Components Claim"]
+[source,cddl]
+----
+tsm-sw-components-label = TBD
+
+tsm-sw-components-type = [
+   tsm-driver
+   tsm
+]
+
+tsm-driver = (riscv-ap-tee-sw-components)
+tsm = (riscv-ap-tee-sw-components)
```

Comment:
It looks like `riscv-ap-tee-sw-components` is not defined

Reply:
Right, I meant `riscv-ap-tee-sw-component`.  Should be fixed now.